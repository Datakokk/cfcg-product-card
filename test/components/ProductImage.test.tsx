import React from "react";
import renderer from "react-test-renderer"
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from "../data/products";


describe('Test on <ProductImage /> ', () => { 
    
    test('show display component without image', () => { 
        

        const wrapper = renderer.create(
            <ProductImage img="someOne.jpg"/>
        );

        expect( wrapper.toJSON() ).toMatchSnapshot();
     });

     test('should display component with image', () => { 

        const wrapper = renderer.create(
            <ProductCard product={ product2 }>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        );
        expect( wrapper.toJSON() ).toMatchSnapshot();
      })
 })