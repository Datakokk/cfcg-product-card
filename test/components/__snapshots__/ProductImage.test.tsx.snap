// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <ProductImage />  should display component with image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Test on <ProductImage />  show display component without image 1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="someOne.jpg"
/>
`;
